app/components/primer/navigation/tab_component.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Primer
+  module Navigation
+    # This component is part of navigation components such as `Primer::TabNavComponent`
+    # and `Primer::UnderlineNavComponent` and should not be used as a standalone component.
+    class TabComponent < Primer::Component
+      include ViewComponent::SlotableV2
+
+      # Panel controlled by the Tab. It will only renderd if `with_panel` is `true`.
```

Comment:
So if we pass a panel, but `with_panel` is `false`, what happens?

Reply:
I think I have to update this description to be more clear
some context here https://github.com/primer/view_components/pull/393/files#r597718804

Reply:
https://github.com/primer/view_components/pull/393/commits/284421bf030150876541299b21a8bc4e7741b772